downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1563 @@
+- The `rabbitmq:Message` object is changed into a record type.
+
+    ```ballerina
+    public type Message record {|
+        byte[] content;
+        string routingKey;
+        string exchange = "";
+        int deliveryTag?;
+        BasicProperties properties?;
+    |};
+    ```
+
+- The `queueDeclare()` function of the `rabbtimq:Channel` is split into `queueDeclare()` and `queueAutoGenerate()` of the `rabbtimq:Client`. 
+- The `basicPublish()` function of the `rabbitmq:Channel` is changed to `publishMessage(Message message)` of the `rabbitmq:Client`. An example of sending a message with the new API is given below.
+
+    ```ballerina
+    import ballerinax/rabbitmq;
+
+    public function main() returns error? {
+        rabbitmq:Client newClient = check new;
+
+        check newClient->queueDeclare("MyQueue");
+
+        string message = "Hello from Ballerina";
+        check newClient->publishMessage({
+            content: message.toBytes(),
+            routingKey: "MyQueue"
+        });
+    }
+    ```
+
+###### Service and Listener Changes
+
+- The return type of the `rabbitmq:Listener init` is changed to `Error?`. 
+- Has a single type of service that supports the three types of remote functions below:
+     - `onMessage(rabbitmq:Message message) {}`
+     - `onMessage(rabbitmq:Message message, rabbitmq:Caller caller) {}`
+     - `onRequest(rabbitmq:Message message) returns anydata {}`
+- A new `onRequest` function is introduced to directly reply to a message by returning a value if the `replyTo` subject is present.
+- If the subject name is not given in the `@rabbitmq:ServiceConfig`, the name of the service is considered as the subject name. 
+- The new syntax with the service changes is given below.
+
+    ```ballerina
+    import ballerinax/rabbitmq;
+
+    listener rabbitmq:Listener channelListener = new;
+
+    @rabbitmq:ServiceConfig {queueName: "MyQueue"}
+    service rabbitmq:Service on channelListener {
+        remote function onMessage(rabbitmq:Message message) {
+        }
+    }
+    ```
+
+##### Time Module Improvements
+
+The improvements below have been added to the `Time` module.
+
+- Added support for commonly-used date-time formatters in the `time:format()` & `time:parse()` APIs.
+- The hours, minutes, seconds, milliseconds, and zone ID parameters of the `time:createTime()` API have been made defaultable.
+- Introduced a new `time:Duration` record type to represent a chunk of time.
+- Modified the `time:addDuration()` and `time:subtractDuration()` APIs to accept the `time:Duration` records to be added/subtracted.
+- Introduced a new `time:getDifference()` API to calculate the difference between two `time:Time` records.
+- Introduced a new `time:getTimezones()` API to retrieve the timezone IDs supported by the underlying native code.
+- Introduced an enum to represent the days of the week and modified the `time:getWeekday()` API to return this enum.
```

Comment:
Should we say method instead of API for each of these? @ThishaniLucas

Reply:
Updated.